### DISCUSSION TITLE
MIB can't find MAP update on SD card

### DISCUSSION THREAD

**Discussion by murat3** [Mr-MIBonk/M.I.B._More-Incredible-Bash#139] [General]
Hello,

SW Train: MHI2_ER_AUG22_K3346

After having successfully upgraded FW and patched my unit, map files can't be found on SD card.
Have tried 3 different microSD cards (one 32GB and two 128GB cards), with different formats (exFAT and FAT) formatted on Mac and Windows but cannot manage to update map files.
Have tried both SD slots and even with USB stick.

FEC codes are all valid. Factory FEC = 02300020
Tried with FEC's: 023000FF 023000EE and 02300030, no succes.
Tried with navi update P310_N60S5MIBH3_EU and P197_N60S5MIBH3_EU.

Current map version v03959802DE 0175 ECE 2019

What am I doing wrong?

**Comment by harman-f** [comment]
I like normal SDs better.
µSD and adpter is always a bit tricky.

For maps you also can use USB Stick.

What exactly - picture please - do you see in SWDL menu, if you try to select the new maps?

**Reply by murat3** [reply]
Via Red Menu, the option to start the update is grayed out.
Via MMI settings menu I select SD card, it scans the card for a while (30-40 seconds) and shows a message saying "did not found any suitable files on card".

After scanning the SD card, the unit puts "WMPInfo.xml" file on the SD card with content as seeen in picture.

<img width="862" alt="image" src="https://user-images.githubusercontent.com/3492326/223842104-d88cee5e-9761-4808-8980-ffccfb4b579f.png">

![20230306_211746 Medium](https://user-images.githubusercontent.com/3492326/223791754-4fe4178d-b1e0-4d55-874a-ae564c3e3168.jpeg)
![20230306_211700 Medium](https://user-images.githubusercontent.com/3492326/223791755-df954fd1-3a03-421d-a6fe-61deec063eb0.jpeg)

![Screenshot 2023-03-08 at 18 52 41 Medium](https://user-images.githubusercontent.com/3492326/223791742-1bb07ca1-3004-4765-8e96-3ab6695097bd.jpeg)
![20230306_211845 Medium](https://user-images.githubusercontent.com/3492326/223791748-01aad918-4430-40f4-98c5-2e9a087bcd80.jpeg)

**Comment by murat3** [comment]
Also tried with two different SD cards (no microSD). But same result.
Anyone an idea how to resolve this issue?

**Comment by harman-f** [comment]
023000FF is not a standard M.I.B patch. 

Share your log file and backup. 

This could be related to wrong coding
Or damaged maps partition.

**Comment by murat3** [comment]
downloaded

Did have same result when on two different (lower) firmwares. Also tried with 023000EE.

**Comment by harman-f** [comment]
Your MU1438 backup looks good:

```RCC-addfec---->
FECs from /net/rcc/mnt/efs-persist/FEC
FECs currently used by unit
023000EE
02300020
00070200
00060900
00060800
00050000
00040100
00030000

FECs from /net/rcc/mnt/efs-system/backup/FEC
FECs from factory
02300020
00070200
00060900
00060800
00050000
00040100
00030000
```

Patch also did run fine.

If you changed it afterwards, restore FECs and revert them back to the satte listet above with ...EE

Please check in GEM - M.I.B. -> NAVIGATION -> NAVIGATION STATUS
And share the picture here.

P310_N60S5MIBH3_EU is correct for EU.
Which country is the car used in?

**Comment by murat3** [comment]
I have restored FEC from backup and added again from patch. Now using 023000EE. Still no luck. Same problem again. Car is a Dutch car in Netherlands.

Before I used MIB to patch, I updated the existing 2019 maps using the online function via Audi Connect. Some country maps were updated. Could this have modified some settings? After this I tried to update the maps using MIB patch. Because it did not work I upgraded the firmware to latest but still same problem.

![20230312_150342 Medium](https://user-images.githubusercontent.com/3492326/224552426-659ab5cd-b653-4e82-869c-dfc3c7c13f84.jpeg)
![20230312_150707 Medium](https://user-images.githubusercontent.com/3492326/224552427-dc99980f-9341-4d8b-84e2-2130d11d6358.jpeg)
![20230312_152438 Medium](https://user-images.githubusercontent.com/3492326/224552428-1d6c1d38-9e3e-48cc-863c-d3b9d5ff55f4.jpeg)
![20230312_152622 Medium](https://user-images.githubusercontent.com/3492326/224552429-61b6bfde-9918-428a-8009-5582d1115b86.jpeg)

**Reply by harman-f** [reply]
Which update menu do you used?

REM?

**Reply by murat3** [reply]
No. MMI standaard update menu.  Haven't tried it via REM.
Should I suppose to use the REM or the MMI menu?

**Reply by harman-f** [reply]
Use REM, the other will not work

**Reply by murat3** [reply]
Just tried via REM. Option still grayed out. Do I have to wait really long for it? Did wait about 3 minutes.
![20230312_163018](https://user-images.githubusercontent.com/3492326/224555076-18950c8a-f887-4688-90ab-094c32aaa69c.jpg)
![20230312_162919](https://user-images.githubusercontent.com/3492326/224555081-94d772b2-3121-4bf2-b176-82a0484df5e2.jpg)
![20230312_162514](https://user-images.githubusercontent.com/3492326/224555085-db8f9444-4266-4f68-b6df-2c4fafcfa768.jpg)
![20230312_162507](https://user-images.githubusercontent.com/3492326/224555087-24d21808-43ad-4c2d-b6b1-adc5bb656da5.jpg)

**Reply by harman-f** [reply]
NavDB you already cleared?

Please show your variant.

M.I.B. -> System Information -> VariantInfo

**Reply by murat3** [reply]
No i did not clear NavDB. I shall try to clear NavDB and then try to update again via REM.

**Reply by murat3** [reply]
How to clear NavDB?

**Reply by harman-f** [reply]
M.I.B. --> Advanced Settings --> Clean Navigation DB

**Reply by murat3** [reply]
After formatting NavDB unit rebooted.
Update via REM sd2, update option grayed out.
Via sd1, update option grayed out.
Via usb, update option grayed out.
Update via MMİ settings, no valid data found.
![20230312_171028](https://user-images.githubusercontent.com/3492326/224558181-7c53c6d2-3f2b-446e-9a32-6a2ad973db9c.jpg)
![20230312_171011](https://user-images.githubusercontent.com/3492326/224558188-b003cf6e-b027-467b-8eff-724ea74f14f6.jpg)

**Comment by harman-f** [comment]
Variant…

**Reply by murat3** [reply]
> Variant…

See picture nr 3 and 2 in previous post for variantinfo

**Reply by harman-f** [reply]
out of ideas on my end.

NAND could be bad...

Do you still see any maps on unit?
Or all are gone now?

**Reply by murat3** [reply]
They are gone now, no navi maps on unit.

**Reply by harman-f** [reply]
Try if someone in MIB2 channel on Telegram has an idea...

Show Fecs, variant, map update error and say that you already deleted navdb

**Reply by murat3** [reply]
Ok, will do.

**Comment by murat3** [comment]
It recognized the maps and is updating now!
Formatting microSD card with FAT32 on windows did the trick. ExFat on windows did not work. Eventually downloaded another tool which could FAT32 format (Windows did not suppport Fat32, only ExFat)

 Thank you very much for your help!

by the way; update took 35 minutes. After 20 mins the ignition turned automatically off but MIB did continue to update. After 34m it rebooted the unit and after 35 mins it finished.

**Reply by Biorn1950** [reply]
Hello, so you have grey update button. Then format your sd card to fat32 and that worked?

**Reply by murat3** [reply]
Yes, be sure you use Windows and choose for FAT32.

**Reply by Biorn1950** [reply]
I encountered the same weird issue.
![image](https://github.com/user-attachments/assets/253c8f6a-18c3-46eb-afb6-2a956c8683ee)
My SDcard was already in fat32 but after:
- Diskpart clean disk
- Diskpart create partition primary
- Format Fat32

All is working fine.

Ps: Disk must be in MBR.

**Reply by jtscarguy** [reply]
> It recognized the maps and is updating now! Formatting microSD card with FAT32 on windows did the trick. ExFat on windows did not work. Eventually downloaded another tool which could FAT32 format (Windows did not suppport Fat32, only ExFat)
> 
> Thank you very much for your help!
> 
> by the way; update took 35 minutes. After 20 mins the ignition turned automatically off but MIB did continue to update. After 34m it rebooted the unit and after 35 mins it finished.

Did you end up using the red or green engineering menu or the default MMI update? I'm having the same issue, when updating through MMI it says no update found, when I use red menu the start update is greyed out.

**Comment by DanielForsell** [comment]
Guys! 

I have absolutely same problem!

I had the maps few days back and the database dissapeared from unit when i opened update for the VC in the RED (Didn’t update). After i still updated the VC and it’s working, no problems.   The car recognices the sd card, but says «No valid update data found on the medium». I tryed multiple time format the sd card and usb stick using macos and windows to Fat32 and exFat. I tryed to extract the maps usin 7zip to pc and directly to the sd card. I have deleted the maps DB using MIB. Factory reseted the unit.  I did once resotre back the factory FEC:s from unit backup and patched the MMI again.

Train info: MHI2_ER_AUG22_K3346
Mu version: 1438 Variant info: FM2-P-TNL-EU-AU-MLE Region: EU Maps trying to install: P390_N60S5MIBH3_EU
FEC: 023000ee

I don’t have any diagnostic tools. 

Need help with this, any ideas or suggestions. 

I'm not sure, but could some activation through obd using lm cause the problems that maps disappeared and i can't now install them.

**Comment by gndworks** [comment]
Hi all. I'm trying to install Australia maps and the start update button was greyed out.

To fix, I had to add the 023D00EE FEC code to the patches folder where other FECs were defined. Then I patches the FECs again by itself in the GEM.

**Reply by debanjanbasu** [reply]
I did the same thing and have the same issue, but still getting maps greyed out...
![95489B91-4F76-4249-A89F-01700AF46403_1_102_o](https://github.com/user-attachments/assets/bce4d7b8-fea7-4bb6-8eb2-e776a5b1b46b)

Any advice?

**Comment by Efe64** [comment]
Hi,
I've tried literally everything but stil getting E15 error codes...

It's an A5 2.0 tfsi 2017. Firmware upgraded to 5092.

- Different sd cards 
- Different formats
- Different map packs
- Rolling back mib settings
- MiB patch

Still no succes... Any suggestions?

**Reply by murat3** [reply]
I had to format the sd card Using Windows and using a tool which supports FAT32; 
Also, use MBR as the partitioning table format.

**Reply by Efe64** [reply]
Unfortunately it didn't work...
I'm almost gonna give up :(

**Reply by Efe64** [reply]
Fixed!

The problem was with the fec codes.

Reinstalled m.i.b., redownloaded patches.

Re patched the car in mib, finally it applied 023000ee.

Thanks for your help

**Comment by murat3** [comment]
Do you have a chance to try the SD card on another Audi?

Did you try this suggestion? https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/discussions/139#discussioncomment-11435122

**Reply by Efe64** [reply]
Also tried that. Gonna try to reinstall mib and patch. I think the problem is with the codes, it's 02300022. It's needs to be ending with ee.

**Comment by momo1072** [comment]
Solved on MHI2_ER_AUG22_K3346 / MU1438 – EU maps

I had a similar problem on an Audi MHI2 unit:

Train: MHI2_ER_AUG22_K3346
MU: 1438
Region: EU / Audi
Map package: 2025/2026 EU (PSF 60)

In the Red Engineering Menu the map package was detected, but I had:

NavDB [F]
Eggnog [E15]
Truffles [E15]
SpeechResVDE [E15]

and “Start Update” was greyed out.

I checked the active FECs with M.I.B. under:

patch_unit_aio -> add_feccontainer.fec -> fec_list

Initially only these were present:

00040100
02310020

The correct patch folder for my unit,

MHI2_ER_AUG22_K3346_MU1438_PATCH

already contained 023000EE in addFecs.txt.

M.I.B. also confirmed:

Unit is already patched!
No need to patch unit again!

Therefore I did not reflash the patch. I used:

patch_unit_aio -> add_feccontainer.fec -> Add new FECs to FecContainer.fec

After reboot I checked fec_list again and 023000EE was present under “FECs in use”.

However, the normal map update still showed NavDB [F] and the Start Update button was still grey.

The step that finally allowed the update was:

M.I.B. -> advanced_settings -> swdl_fw_updates -> Enable user defined SWDL

I left “Ignore Region and Variant” disabled.

Then I returned to the Red Engineering Menu, selected User Defined, opened NavDB and manually selected the NavDB components. NavDB changed to [S] and Start Update became selectable.

I started the update and the EU regions installed one after another. The process completed with:

NavDB [OK]

After returning to the normal MMI, the 2025/2026 EU maps work correctly, address search works and route calculation works.

Hopefully this helps anyone with K3346/MU1438 + NavDB [F]/E15 + greyed-out Start Update.
